src/server/routes/transaction/blockchain/getLogs.ts

```diff
@@ -0,0 +1,189 @@
+
+declare global {
+  interface BigInt {
+    toJSON: () => string;
+  }
+}
+BigInt.prototype.toJSON = function () {
+  return `${this}n`;
+};
+
+// INPUT
+const requestSchema = Type.Object({
+  txHash: Type.String({
+    description: "Transaction hash",
+    examples: [
+      "0xd9bcba8f5bc4ce5bf4d631b2a0144329c1df3b56ddb9fc64637ed3a4219dd087",
+    ],
+    pattern: "^0x([A-Fa-f0-9]{64})$",
+  }),
+  chain: Type.String({
+    examples: ["80002"],
+    description: "Chain ID or name",
+  }),
+});
+
+// OUTPUT
+export const responseBodySchema = Type.Object({
+  result: Type.Array(
+    Type.Object({
+      eventName: Type.String(),
+      args: Type.Unknown({
+        description:
+          "Event arguments. These are mostly key-value pairs, with values being numbers or strings. BigInts are represented as strings suffixed with 'n', eg: '1000000000000000000n'",
+        examples: [
+          {
+            from: "0xdeadbeeefdeadbeeefdeadbeeefdeadbeeefdead",
+            to: "0xdeadbeeefdeadbeeefdeadbeeefdeadbeeefdead",
+            value: "1000000000000000000n",
+          },
+        ],
+      }),
+      address: Type.String(),
+      topics: Type.Array(Type.String()),
+      data: Type.String(),
+      // this is a lie, we say bigint but we actually return a string
+      blockNumber: Type.BigInt({
+        description:
+          "Block number where the event was emitted, this is a BigInt encoded as a string suffixed with 'n'",
+        examples: ["79326434n"],
+      }),
+      transactionHash: Type.String(),
+      transactionIndex: Type.Number(),
+      blockHash: Type.String(),
+      logIndex: Type.Number(),
+      removed: Type.Boolean(),
+    }),
+  ),
+});
+
+responseBodySchema.example = {
+  result: [
+    {
+      eventName: "Transfer",
+      args: {
+        from: "0x0000000000000000000000000000000000000000",
+        to: "0x71B6267b5b2b0B64EE058C3D27D58e4E14e7327f",
+        value: "1000000000000000000n",
+      },
+      address: "0x71b6267b5b2b0b64ee058c3d27d58e4e14e7327f",
+      topics: [
+        "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0x00000000000000000000000071b6267b5b2b0b64ee058c3d27d58e4e14e7327f",
+      ],
+      data: "0x0000000000000000000000000000000000000000000000000de0b6b3a7640000",
+      blockNumber: "79326434n",
+      transactionHash:
+        "0x568eb49d738f7c02ebb24aa329efcf10883d951b1e13aa000b0e073d54a0246e",
+      transactionIndex: 1,
+      blockHash:
+        "0xaffbcf3232a76152206de5f6999c549404efc76060a34f8826b90c95993464c3",
+      logIndex: 0,
+      removed: false,
+    },
+  ],
+};
+
+export async function getTxLogs(fastify: FastifyInstance) {
+  fastify.route<{
+    Params: Static<typeof requestSchema>;
+    Reply: Static<typeof responseBodySchema>;
+  }>({
+    method: "GET",
+    url: "/transaction/:chain/logs/:txHash",
```

Comment:
Engine is really hard to be RESTful since it does so much outside of CRUD.

This URL path is pretty unintuitive anyway. WDYT about just using query params:
`GET /transaction/logs?chainId=8453&hash=0x...`